### PR TITLE
6 hw6

### PR DESCRIPTION
**Pull request by jGrzyb**

### CODE REVIEW

**Comment by jGrzyb** [comment]
I'm sorry, but I don't understand the comment nor task 7... Doesn't factory method require catalog interface and eCatalog, PaperCatalog inheriting from it? How are we supposed to implement it on single catalog if both have completely different method of getting data from file?

File: Hw5/Task2/Author.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
+using System.Text.RegularExpressions;
+
```
Comment (IgorKheidorov) [inline Hw5/Task2/Author.cs:2]: Remove

File: Hw5/Task2/Author.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        Name = name;
+        Surname = surname;
+        BirthDate = birthDate;
+    }
+
+    public override string ToString() 
+    {
+        return $"{Name} {Surname}";
+    }
+
```
Comment (IgorKheidorov) [inline Hw5/Task2/Author.cs:63]: Check name

File: Hw5/Task2/Author.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
+using System.Text.RegularExpressions;
+
+[Serializable]
+public class Author 
+{
+    private string _name = "";
+    private string _surname = "";
+    private DateTime? _birthDate = null;
+
```
Comment (IgorKheidorov) [inline Hw5/Task2/Author.cs:9]: ChwckName(value);
name= value.

File: Hw5/Task2/Author.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        BirthDate = birthDate;
+    }
+
+    public override string ToString() 
+    {
+        return $"{Name} {Surname}";
+    }
+
+    private string setName(string value)
+    {
```
Comment (IgorKheidorov) [inline Hw5/Task2/Author.cs:65]: const int max_length = 200;

File: Hw5/Task2/Catalog.cs [hunk 5]
```diff
@@ -1,9 +1,11 @@
-using System.Collections.ObjectModel;
-using System.Text.RegularExpressions;
-
 public class Catalog
 {
-    Dictionary<Isbn, Book> dictionary = new();
+    public Dictionary<Isbn, Book> dictionary = new();
+
```
Comment (IgorKheidorov) [inline Hw5/Task2/Catalog.cs:4]: remove all empmty consructors

File: Hw5/Task2/JsonRepository.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        DALCatalogAuthors authorsWithBooks = new();
+        foreach (var file in Directory.EnumerateFiles(filePath))
+        {
+            var json = File.ReadAllText(file);
+            var entry = JsonSerializer.Deserialize<DALAuthorEntry>(json);
+            if(entry == null)
+            {
+                Console.WriteLine("-------- Error: Could not deserialize file: " + file);
```
Comment (IgorKheidorov) [inline Hw5/Task2/JsonRepository.cs:22]: Remove

**Review by IgorKheidorov — APPROVED** [review]
Please fix and merge

**Pull request merged**